### PR TITLE
Re-organize directory structure

### PR DESCRIPTION
**Pull request by ilevkivskyi**
See discussion in https://github.com/python/typeshed/issues/2491

This PR needs a mypy commit that supports new directory structure (including `--custom-typeshed-dir` for standard library) landed in mypy master, which may take another day or two.

cc @JukkaL

### CODE REVIEW

**Comment by ilevkivskyi** [comment]
OK, here is the situation: all tests pass, but I temporarily disabled two tests:
* Test stubs with `pytype`
* Run `stubtest` on stubs
Somehow `mypy_primer` passes, but it will still needs to be updated to support this.

This is now ready for review, I am aiming to land this tomorrow morning.

**Review by JelleZijlstra — APPROVED** [review]

File: tests/mypy_test.py [hunk 1]
```diff
@@ -98,44 +140,58 @@ def main():
...
-                                    files.append(fn)
+                add_files(files, seen, root, name, args, exclude_list)
+        else:
+            supported_versions = parse_versions(os.path.join("stdlib", "VERSIONS"))
+            root = "stdlib"
+            for name in os.listdir(root):
+                if name == PY2_NAMESPACE:
+                    continue
+                mod, _ = os.path.splitext(name)
+                if mod not in supported_versions or supported_versions[mod] > (major, minor):
```
Comment (srittau) [inline tests/mypy_test.py:161]: Should we print a warning or even fail the test if `mod` is not in `supported_versions`?

**Review by srittau — APPROVED** [review]
Thanks! LGTM, with one question.

**Review by rchen152 — APPROVED** [review]

**Review by JukkaL — APPROVED** [review]

File: tests/mypy_test.py (hunk 1 above)
Comment (ilevkivskyi) [reply tests/mypy_test.py:161]: Yes, I think we should, I was thinking about allowing some unrelated files in that directory (like some scripts etc). But we actually should keep directory structure clean. I will update this and will also add in a separate PR later a test that ensures `stdlib` and `stubs` have strict directory structure.

**Comment by rchen152** [comment]
I believe I have pytype_test working again; it requires first a change to pytype itself (https://github.com/google/pytype/pull/821), then an update to the test (https://github.com/python/typeshed/commit/2de4e0200ced919a02283ddfea12ee52864302d8). (It'll take at least a day for the pytype change to be reviewed and merged, so it still makes sense to disable pytype_test first and re-enable it later.)

**Comment by ilevkivskyi** [comment]
@rchen152 Great, thanks! I will then merge this now with pytype test disabled, and you can re-enable it whenever ready.

**Comment by Akuli** [comment]
Each stub now shows up with "1 contributor" on github, but if I look at the blame (by pressing `b`), I can see all contributors.

**Comment by ilevkivskyi** [comment]
> Each stub now shows up with "1 contributor" on github, but if I look at the blame (by pressing b), I can see all contributors.

This is a GitHub bug I think `git blame` works correctly in this case (since these are pure file renames).

**Pull request merged**